pyccel/parser/syntactic.py

```diff
@@ -786,12 +786,14 @@ def fill_types(ls):
         results = []
         result_counter = 1
 
+        local_symbols = self.scope.local_used_symbols
+
         for r in zip(*returns):
             r0 = r[0]
 
             pyccel_symbol  = isinstance(r0, PyccelSymbol)
             same_results   = all(r0 == ri for ri in r)
-            name_available = all(r0 != a.name for a in arguments)
+            name_available = all(r0 != a.name for a in arguments) and r0 in local_symbols
```

Comment:
just asking if i understood correctly.

we are checking if we have the name in the `local_symbols` if yes then just return it without creating a `tmp` variable?

Reply:
Yes exactly. For example:
```python
def f():
    a = 4
    return a
```
`a` is a local symbol so we don't need to do:
```python
def f():
    a = 4
    Out_0001 = a
    return Out_0001
```
to ensure that a locally declared variable is returned